### ISSUE TITLE
Maximum and Minimum number of terms per post

### ISSUE THREAD

**Issue by stevejburge** [TaxoPress/TaxoPress#1746]
### Discussed in https://github.com/TaxoPress/TaxoPress/discussions/1719

<div type='discussions-op-text'>

<sup>Originally posted by **stevejburge** June 19, 2023</sup>
Related to this: https://github.com/TaxoPress/TaxoPress/issues/1676

We could add a setting for the maximum and Minimum number of terms per post</div>

**Comment by ojopaul** [comment]
@stevejburge Can you confirm the area this settings applied to?

1. Post TaxoPress Metabox
2. Metaboxes Preview (Add Term button)
3. Existing Content
4. Auto Term Schedule
5. ???

**Comment by ojopaul** [comment]
@stevejburge Can you confirm the area this settings applied to?

1. Post TaxoPress Metabox
2. Metaboxes Preview (Add Term button)
3. Existing Content
4. Auto Term Schedule
5. ???

I'll like to mention this does not perform any strict check for post or prevent post from been updated but instead, this is for only TaxoPress taxonomy area prevent user from selecting maximum post.

Also, the minimum terms per post seems useless to me in most of those area especially on the post screen as we're not limiting post publish or preventing the post from been saved. So, ideally, no way for us to restrict the minimum post terms.

If you ask me, I'll say minimum is not really so useful here unless we want to import whole checklist feature feature for preventing post publish if post doesn't have certain number of taxonomy terms and that raise a question on the usefulness of the minimum settings

**Comment by stevejburge** [comment]
Hi @ojopaul. This is strictly for the Metaboxes feature.

Understood on the limitations. I can add that to the documentation. This can simply be a visual restriction.

**Comment by ojopaul** [comment]
You still didn't answer my concern on the usefulness of minimum terms.

For example, if minimum term is set to 2, and I'm editing a post, how does this work or help visual restriction on the following metabox option?
- Default
- Dropdown
- Radio
- Checkbox


How exactly does the minimum works here and how is it implemented?

![mini](https://github.com/user-attachments/assets/e2aae50b-5222-43d5-8e54-f0414d6b9c85)
@stevejburge

**Comment by stevejburge** [comment]
Can we revisit this for the checkboxes option, and perhaps the dropdown option?

**Comment by ojopaul** [comment]
@stevejburge Do you have answer to my last question/responses on this issue now?

**Comment by ojopaul** [comment]
@stevejburge I dropped this comment before on this thread we postponed this issue https://github.com/TaxoPress/TaxoPress/issues/1746#issuecomment-2588828711

If you have a suggestion on how exactly you think the minimum will work, happy to implement it.

> You still didn't answer my concern on the usefulness of minimum terms.
> 
> For example, if minimum term is set to 2, and I'm editing a post, how does this work or help visual restriction on the following metabox option?
> 
> * Default
> * Dropdown
> * Radio
> * Checkbox
> 
> How exactly does the minimum works here and how is it implemented?
> 
> ![mini](https://github.com/user-attachments/assets/e2aae50b-5222-43d5-8e54-f0414d6b9c85) [@stevejburge](https://github.com/stevejburge)

**Comment by stevejburge** [comment]
Thanks @ojopaul. Could we treat this like the PublishPress Checklists plugin?
https://publishpress.com/knowledge-base/number-of-tags-or-taxonomy-terms/

![Image](https://github.com/user-attachments/assets/ab0d2439-3270-4280-83b5-9dd285e58826)
![Image](https://github.com/user-attachments/assets/e0716ae7-8126-4a72-ae44-b6a990a100d2)

**Comment by ojopaul** [comment]
@stevejburge That brought back my old question which never receive a response on this conversion.

Are we agreeing to implementing Checklists like feature which include Blocking Publish button or publishing in both classic editor and Gutenberg, adding a sidebar metabox for this restriction as well dealing with possible future conflict that may arise with this feature?

**Comment by stevejburge** [comment]
@ojopaul Fair point. OK, let's allow this one to drop. We can point people to PublishPress Checklists if they need this feature.
